lib/connectors_async/job_runner.rb

```diff
@@ -0,0 +1,44 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'concurrent'
+
+require 'connectors_async/job_store'
+require 'connectors_shared/job_status'
+
+module ConnectorsAsync
+  class JobRunner
+    def initialize
+      @pool = Concurrent::ThreadPoolExecutor.new(min_threads: 1, max_threads: 4, max_queue: 0)
```

Comment:
We'll want to make `max_threads` configurable. I don't have a strong opinion as to which PR that falls under.

Reply:
Good call, I've made an attempt to do so in 85f3d011ef85a0b813e8d39374ffeba26b40ce36